plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/ProxyServiceDeserializer.java

```diff
@@ -100,6 +100,14 @@
 			hasPublishWsdl=false;
 		}
 		
+		if (object.getPreservePolicy() != null) {
```

Comment:
We can avoid this null check 

"true".equals.(object.getPreservePolicy())

Reply:
This null check is needed because if this attribute gets removed from the source view then when saving this becomes null

Reply:
Okay, If you change the code with the suggested way NPE won't occur.

Reply:
Thanks. In order to persist the default value as 'true' even it gets removed from the source view, will change the code to
                if("false".equals(object.getPreservePolicy())){
			executeSetValueCommand(PROXY_SERVICE__PRESERVE_POLICY, false);
		} else {
			executeSetValueCommand(PROXY_SERVICE__PRESERVE_POLICY, true);
		}